app/interactors/comments/save.rb

```diff
@@ -0,0 +1,12 @@
+module Comments
+  class Save
+    include Interactor
+
+    delegate :comment, to: :context
+
+    def call
+      context.fail!(error: "Invalid comment details") unless comment.valid?
+      comment.save
```

Comment:
Two action instead of one

```suggestion
      context.fail!(error: "Invalid comment details") unless comment.save
```